src/gui/qgsconfigureshortcutsdialog.cpp

```diff
@@ -262,22 +269,44 @@ void QgsConfigureShortcutsDialog::loadShortcuts()
     currentLocale = QLocale().name();
   }
 
+  const QString version = root.attribute( QStringLiteral( "version" ) );
+
   if ( root.attribute( QStringLiteral( "locale" ) ) != currentLocale )
   {
-    QMessageBox::information( this, tr( "Loading Shortcuts" ),
-                              tr( "The file contains shortcuts created with different locale, so you can't use it." ) );
-    return;
+    if ( QgsProjectVersion( version ) < QgsProjectVersion( QStringLiteral( "1.1" ) ) )
+    {
+      QMessageBox::information( this, tr( "Loading Shortcuts" ),
+                                tr( "The file contains shortcuts created with different locale, so you can't use it." ) );
+      return;
+    }
+    else // From version 1.1, if objectName is not empty, it is used as key.
+    {
+      QMessageBox::information( this, tr( "Loading Shortcuts" ),
+                                tr( "The file contains shortcuts created with different locale, so some shortcuts may not work." ) );
+    }
   }
 
   QString actionName;
   QString actionShortcut;
+  QString actionSettingKey;
 
   QDomElement child = root.firstChildElement();
   while ( !child.isNull() )
   {
-    actionName = child.attribute( QStringLiteral( "name" ) );
     actionShortcut = child.attribute( QStringLiteral( "shortcut" ) );
-    mManager->setKeySequence( actionName, actionShortcut );
+    if ( version < "1.1" )
```

Comment:
This should use QgsProjectVersion for the comparison too

Reply:
My bad. Done.